src/effects/effectchain.cpp

```diff
@@ -200,23 +200,26 @@ const QString& EffectChain::presetName() const {
     return m_presetName;
 }
 
-void EffectChain::loadChainPreset(EffectChainPresetPointer pPreset) {
+void EffectChain::loadChainPreset(EffectChainPresetPointer pChainPreset) {
     slotControlClear(1);
-    VERIFY_OR_DEBUG_ASSERT(pPreset) {
+    VERIFY_OR_DEBUG_ASSERT(pChainPreset) {
         return;
     }
 
-    int effectSlotIndex = 0;
-    for (const auto& pEffectPreset : pPreset->effectPresets()) {
-        EffectSlotPointer pEffectSlot = m_effectSlots.at(effectSlotIndex);
-        pEffectSlot->loadEffectFromPreset(pEffectPreset);
-        effectSlotIndex++;
+    const QList effectPresets = pChainPreset->effectPresets();
+
+    // TODO: use C++23 std::ranges::views::zip instead
+    const std::size_t validPresetSlotCount = std::min(effectPresets.count(), m_effectSlots.count());
+    for (std::size_t presetSlotIndex = 0;
```

Comment:
QT uses int for sized from Qt6 it is qsizetype which is ssize_t
```suggestion
    const int validPresetSlotCount = std::min(effectPresets.count(), m_effectSlots.count());
    for (int presetSlotIndex = 0;
```
This avoids a warning in MSVC

Reply:
done